tests/contracts/blsContracts/BLSLibrary.sol

```diff
@@ -0,0 +1,235 @@
+        bytes memory _dst,
+        uint16 count
+    ) private pure returns (bytes memory) {
+        require(_dst.length == 0, "DST not implemented yet");
+
+        //require(l <= type(uint16).max, "length exceeds uint16 range");
+        uint16 length = L * count;
+        uint64 ell = (length + 31) / 32;
+        bytes32 zero;
+        uint8 zbyte;
+        uint8 one = 1;
+
+        // Z_pad = I2OSP(0, r_in_bytes)   64
+        // l_i_b_str = I2OSP(len_in_bytes, 2)
+        // DST_prime = DST ∥ I2OSP(len(DST), 1)
+
+        // b₀ = H(Z_pad ∥ msg ∥ l_i_b_str ∥ I2OSP(0, 1) ∥ DST_prime)
+        bytes32 b0 = sha256(
+            abi.encodePacked(zero, zero, message, length, zbyte, zbyte)
+        );
+        // b₁ = H(b₀ ∥ I2OSP(1, 1) ∥ DST_prime)
+        bytes32 b1 = sha256(abi.encodePacked(b0, one, zbyte));
+
+        require(b1 != sha256(abi.encodePacked(message)));
+
+        bytes memory res = abi.encodePacked(b1);
+        bytes32 strxor;
+        for (uint8 i = 2; i <= ell; i++) {
+            // b_i = H(strxor(b₀, b_(i - 1)) ∥ I2OSP(i, 1) ∥ DST_prime)
+            strxor = b0 ^ b1;
+            b1 = sha256(abi.encodePacked(strxor, i, zbyte));
+            res = abi.encodePacked(res, b1);
+        }
+        return res;
+    }
+
+    function hashToFieldPoints(bytes memory message)
+        private
+        view
+        returns (Elements.Element[] memory)
+    {
+        bytes memory pseudoRandomBytes = hash(
+            abi.encodePacked(message),
+            abi.encodePacked(dst),
+            elCount
+        );
+
+        require(pseudoRandomBytes.length != 0, "No data");
+
+        Elements.Element[] memory res = new Elements.Element[](elCount);
+
+        BigNumber memory modulo;
+        modulo = BigNumbers.init(moduloBytes, false);
+
+        // for number of field points points
+        for (uint256 i = 0; i < elCount; i++) {
+            bytes memory chunk = new bytes(64);
+            uint256 chunkStart = i * 64;
+
+            assembly {
+                let chunkPtr := add(chunk, 0x20) // Points to the start of the `chunk` data
+                let srcPtr := add(add(pseudoRandomBytes, 0x20), chunkStart) // Points to the correct position in `pseudoRandomBytes`
+                mstore(chunkPtr, mload(srcPtr)) // Copy 32 bytes
+                mstore(add(chunkPtr, 0x20), mload(add(srcPtr, 0x20))) // Copy the next 32 bytes
+            }
+
+            BigNumber memory v = BigNumbers.init(chunk, false);
+            res[i] = setBigIntToElement(v, modulo);
+        }
+
+        return res;
+    }
+
+    function setBigIntToElement(BigNumber memory v, BigNumber memory modulo) private view returns (Elements.Element memory) {
+        
+        // returning Elements.toMont(Elements.ElementFromBytes(vv.val));  //this is working to get correct field point but with Montgomery reduction
+        // returning Elements.ElementFromBytes(v.val); // correct field point without Montgomery reduction, but working for MapToG2
```

Comment:
This seems to contain some important insights, but in its current form it looks like obsolete code that should be removed. Can you convert the insights covered in the comments into some code comment for future reference?